metadata.yaml

```diff
@@ -18,7 +18,7 @@ resources:
   zookeeper-image:
     type: oci-image
     description: OCI Image for Apache ZooKeeper
-    upstream-source: ubuntu/zookeeper:latest
+    upstream-source: dataplatformoci/zookeeper:3.6.3
 
```

Comment:
I have seen [here](https://hub.docker.com/r/dataplatformoci/zookeeper/tags) that there is a less recent (11 days ago) but higher version (3.7.1) available. 

Some of the code in this branch was already in sync with 3.6.3 so I decided to go for that.

Reply:
yes, the charm is currently only tested against 3.6.3, but in the future imo we should track each of the supported versions with the rock and charm

Reply:
@marcoppenheimer i think it is required at this time because of how this is refactored.  The PR should really be called something else cause i dont think its particularly representative of what is happening here

Reply:
Let's try out the 3.7.1! If CI/CD works, we can keep the higher version. Otherwise, we'll see what breaks and we can add an item in the backlog. 

Depending on this, I will rename the PR accordingly

Reply:
CI/CD Pipelines fails on 3.7.1. There seems to be some problem with tls tests, at the very least. I believe that ought to be properly addressed in a separate ticket